cartoframes/client/sql_client.py

```diff
@@ -0,0 +1,208 @@
+          instance can be used in place of a `base_url`/`api_key` combination.
+        session (requests.Session, optional): requests session. See `requests
+          documentation
+          <http://docs.python-requests.org/en/master/user/advanced/>`__
+          for more information.
+
+    Example:
+
+        .. code::
+
+            from cartoframes.auth import Credentials
+            from cartoframes.client import SQLClient
+
+            creds = Credentials(username='<YOUR USER NAME>', api_key='<YOUR API KEY>')
+            sql_client = SQLClient(creds)
+
+            sql_client.query('SELECT * FROM table_name')
+            sql_client.execute('DROP TABLE table_name')
+
+            sql_client.distinct('table_name', 'column_name')
+            sql_client.count('table_name')
+            ...
+    """
+
+    def __init__(self, credentials, session=None):
+        self._is_org_user = None
+        self._creds = credentials
+        self._client = create_client(credentials, session)
+
+    def query(self, query, verbose=False):
+        """Run a SQL query. It returns a JSON object with the response.
+        If the `verbose` params is True it returns the full SQL response.
+        """
+        try:
+            response = self._client.execute_query(query)
+            if not verbose:
+                return response.get('rows')
+            else:
+                return response
+        except CartoException as e:
+            print('Error: {}'.format(e))
+
+    def execute(self, query):
+        """Run a long running query. It returns an object with the
+        status and information of the job."""
+        try:
+            return self._client.execute_long_running_query(query)
+        except CartoException as e:
+            print('Error: {}'.format(e))
+
+    def distinct(self, table_name, column_name):
+        """Get the distict values and their count in a table
+        for a specific column."""
+        query = '''
+            SELECT {0}, COUNT(*) FROM {1}
+            GROUP BY 1 ORDER BY 2 DESC
+        '''.format(column_name, table_name)
+        output = self.query(query)
+        return list(map(lambda x: (x.get(column_name), x.get('count')), output))
+
+    def count(self, table_name):
+        """Get the number of elements of a table."""
+        query = 'SELECT COUNT(*) FROM {};'.format(table_name)
+        output = self.query(query)
+        return output[0].get('count')
+
+    def bounds(self, query):
+        """Get the bounds of the geometries in a table."""
+        output = self.query('''
+            SELECT ARRAY[
+                ARRAY[st_xmin(geom_env), st_ymin(geom_env)],
+                ARRAY[st_xmax(geom_env), st_ymax(geom_env)]
+            ] bounds FROM (
+                SELECT ST_Extent(the_geom) geom_env
+                FROM ({}) q
+            ) q;
+        '''.format(query))
+        return output[0].get('bounds')
+
+    def schema(self, table_name, raw=False):
```

Comment:
I think this can be written more concisely and Pythonic-ly with dict/list comprehensions. Appending list elements is slower than comprehensions too.
```suggestion
    def schema(self, table_name, raw=False):
        """Show information about the schema of a table.
        Setting raw=True is returns a JSON with the data."""
        query = 'SELECT * FROM {0} LIMIT 0'.format(table_name)
        output = con.sql_client.send(query, verbose=True)
        fields = output.get('fields')
        if raw:
            return {key: fields[key]['type'] for key in fields}
        else:
            rows = [(key, fields[key]['type']) for key in fields]
            self._print_table(rows, columns=['Column name', 'Column type'], padding=[10, 5])
```